### ISSUE TITLE
Map should auto size to fit all posts

### ISSUE THREAD

**Issue by jshorland** [ushahidi/platform#1411]
This was a feature spec'd in mustang but not yet built and integrated. to fix #1339 properly as @brandonrosage, we need to implement this. This is still lower priority than embeds, location search, and free form survey builds in my opinion.

@rjmackay given this is already scoped pretty clearly in the mustang patterns, can I throw this to you for a first pass for sizing? and/or letting me know additional details/questions you'll need.

**Comment by rjmackay** [comment]
> given this is already scoped pretty clearly in the mustang patterns, can I throw this to you for a first pass for sizing

.. erm can you summarize what the change is? I'm not at all clear what is scoped in mustang but not implemented.

**Comment by rjmackay** [comment]
FYI the map currently auto sizes to fill the screen, and auto *centers* to show all posts. 
If it doesn't thats a bug, though quite possible given we've broken this code a couple of times.

**Comment by jshorland** [comment]
Got it. Apologies for incorrect verbage in my first description. The map should auto center and zoom to fit all posts displayed on the map in any deployment. Currently, the map centers and zooms based on settings in "general settings". I'm going to double check this, but on quakemap.ushahidi.io, it does not auto center and zoom to include all posts (zoom out and you'll see a ton of posts on the north pole. greedy santa).

**Comment by brandonrosage** [comment]
@jshorland is right. The change is effectively to make the map in "Map" mode *always* use Leaflet's [`fitBounds`](http://leafletjs.com/reference.html#map-set-methods). When loading a dataset in "Map" mode, the map should just center and zoom to "the maximum zoom level possible" for the geographical bounds of the markers in context.

**Comment by rjmackay** [comment]
Ok. In that case this is actually just a bug. I think our mapping code need a refactor because this keeps breaking. I suggest we take the opportunity to remove `angular-leaflet-directive` and simply use leaflet directly.. that should be much less error prone.

**Comment by ClearLakeHABMap** [comment]
I would seek the opposite action upon opening the map. By negating the zoom level on the General setting and fitting to zoom, the current situation disregards the geographic area of concern. An example of this would be when you have no data points in Maine and Florida, so those points are zoomed out of a USA based map. Other maps may see more extreme loss of coverage and could discourage participants from using the map as it is not visible upon opening the site. I believe that the option to zoom to fit all entries should be in Settings>General and the zoom level could be overrode by this selection.
